crs_motion_planning/src/path_planning_utils.cpp

```diff
@@ -109,300 +109,389 @@ bool crsMotionPlanner::generateSurfacePlans(pathPlanningResults::Ptr& results)
       {
-        // Generate Descartes preplan
         if (split_strip.poses.size() >= config_->minimum_raster_length)
         {
-          geometry_msgs::msg::PoseArray split_strip_ar;
-
-          if (config_->add_approach_and_retreat)
-          {
-            crs_motion_planning::addApproachAndRetreat(
-                split_strip, config_->approach_distance, config_->retreat_distance, split_strip_ar);
-          }
-          else
-          {
-            split_strip_ar = split_strip;
-          }
-          std::cout << "DESCARTES ROUND 2" << std::endl;
           results->reachable_waypoints.poses.insert(
               results->reachable_waypoints.poses.end(), split_strip.poses.begin(), split_strip.poses.end());
-          if (generateDescartesSeed(split_strip_ar, failed_edges, failed_vertices, joint_traj_msg_out_final))
-          {
-            split_traj.push_back(joint_traj_msg_out_final);
-            split_reachable_rasters.push_back(std::move(split_strip_ar));
-            any_successes = true;
-            std::cout << "SUCCESS" << std::endl;
-          }
-          else
-          {
-            results->unreachable_waypoints.poses.insert(
-                results->unreachable_waypoints.poses.end(), split_strip_ar.poses.begin(), split_strip_ar.poses.end());
-            std::cout << "ADDED APPROACH AND/OR RETREAT MADE RASTER UNREACHABLE" << std::endl;
-          }
+          split_reachable_rasters.push_back(std::move(split_strip));
+          any_successes = true;
         }
         else
         {
           results->skipped_rasters.push_back(split_strip);
         }
-        std::cout << "DONE" << std::endl;
       }
+      std::cout << "DONE" << std::endl;
     }
     else
     {
-      std::cout << "SUCCESS" << std::endl;
-      geometry_msgs::msg::PoseArray split_strip_ar;
-      if (config_->add_approach_and_retreat)
+      if (strip.poses.size() >= config_->minimum_raster_length)
       {
-        crs_motion_planning::addApproachAndRetreat(
-            strip, config_->approach_distance, config_->retreat_distance, split_strip_ar);
+        std::cout << "STRIP FULLY REACHABLE" << std::endl;
+        results->reachable_waypoints.poses.insert(
+            results->reachable_waypoints.poses.end(), strip.poses.begin(), strip.poses.end());
+        split_reachable_rasters.push_back(std::move(strip));
+        any_successes = true;
       }
       else
       {
-        split_strip_ar = strip;
+        results->skipped_rasters.push_back(strip);
       }
-      results->reachable_waypoints.poses.insert(
-          results->reachable_waypoints.poses.end(), strip.poses.begin(), strip.poses.end());
-      std::cout << "DESCARTES ROUND 2" << std::endl;
-      if (generateDescartesSeed(split_strip_ar, failed_edges, failed_vertices, joint_traj_msg_out_final))
+    }
+  }
+  if (!any_successes)
+  {
+    std::cout << "NO REACHABLE POINTS FOUND OR ALL REACHABLE STRIPS ARE TOO SHORT" << std::endl;
```

Comment:
You could store the string into results->msg_out and then print it to the console using that variable